src/main/java/me/rigamortis/seppuku/impl/module/combat/NoCrystalModule.java

```diff
@@ -29,83 +30,73 @@
     }
 
     @Listener
     public void onWalkingUpdate(EventUpdateWalkingPlayer event) {
-        if (event.getStage() == EventStageable.EventStage.PRE) {
-            final FreeCamModule freeCam = (FreeCamModule) Seppuku.INSTANCE.getModuleManager().find(FreeCamModule.class);
-
-            if(freeCam != null && freeCam.isEnabled()) {
-                return;
-            }
+        final boolean instant = placeDelay.getValue() == 0;
+        if (this.placeTimer.passed(this.placeDelay.getValue()) || instant) {
+            if (event.getStage() == EventStageable.EventStage.PRE) {
+                final FreeCamModule freeCam = (FreeCamModule) Seppuku.INSTANCE.getModuleManager().find(FreeCamModule.class);
 
-            final Vec3d pos = MathUtil.interpolateEntity(mc.player, mc.getRenderPartialTicks());
-            final float playerSpeed = (float) MathUtil.getDistance(pos, mc.player.posX, mc.player.posY, mc.player.posZ);
+                if (freeCam != null && freeCam.isEnabled()) {
+                    return;
+                }
 
-            final BlockPos interpPos = new BlockPos(pos.x, pos.y, pos.z);
+                final Vec3d pos = MathUtil.interpolateEntity(mc.player, mc.getRenderPartialTicks());
+                final float playerSpeed = (float) MathUtil.getDistance(pos, mc.player.posX, mc.player.posY, mc.player.posZ);
 
-            final BlockPos north = interpPos.north();
-            final BlockPos south = interpPos.south();
-            final BlockPos east = interpPos.east();
-            final BlockPos west = interpPos.west();
+                final BlockPos interpPos = new BlockPos(pos.x, pos.y, pos.z);
 
-            final BlockPos northBelow = interpPos.north().down();
-            final BlockPos southBelow = interpPos.south().down();
-            final BlockPos eastBelow = interpPos.east().down();
-            final BlockPos westBelow = interpPos.west().down();
+                final BlockPos north = interpPos.north();
+                final BlockPos south = interpPos.south();
+                final BlockPos east = interpPos.east();
+                final BlockPos west = interpPos.west();
 
-            int lastSlot;
-            final int slot = findStackHotbar(Blocks.OBSIDIAN);
-            if (hasStack(Blocks.OBSIDIAN) || slot != -1) {
-                if ((mc.player.onGround && playerSpeed <= 0.005f)
-                        && (this.sneak.getValue() || (!mc.gameSettings.keyBindSneak.isKeyDown()))) {
+                final BlockPos[] surroundBlocks = {north.down(), south.down(), east.down(), west.down(),
+                        north, south, east, west};
 
-                    lastSlot = mc.player.inventory.currentItem;
-                    mc.player.inventory.currentItem = slot;
-                    mc.playerController.updateController();
+                int lastSlot = 0;
+                final int slot = findStackHotbar(Blocks.OBSIDIAN);
+                if (hasStack(Blocks.OBSIDIAN) || slot != -1) {
```

Comment:
Isn't this really redundant? You are finding a slot index with obsidian firstly, then checking if you are holding it. Regardless of what either outcome is, the same thing happens. If you were to not do all of the item swapping related code for when you have it in your hand (it assigns itself the same value so why do anything else) it would make sense.

Reply:
Yeah, I guess that's pretty redundant. Fixed.